src/components/editor/EditorLogin.jsx

```diff
@@ -51,6 +85,29 @@ export default class Login extends BaseComponent {
     }
   }
 
+  onGoogleResponse(response) {
+    if (response.error) {
+      alert(`Google Login Failed.\n${response.error}`);
+      return;
+    }
+
+    const token = response.tokenId;
+
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/googlelogin', false);
+    xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+        browserHistory.push('/');
+      } else {
```

Comment:
Hmm this didn't happen when I just tested it before, but if `onreadystatechange` was fired during any of the none `DONE` readyStates wouldn't this fire? Let's make this `else if (xhr.readyState === XMLHttpRequest.DONE)` instead, yeah? Unless I missed something?

Reply:
I actually made the XMLHttpRequest synchronous with `xhr.open('POST', '/googlelogin', false);` - logically the user has to wait till the login finishes to do anything else. 
when it was async this kept firing indeed. Should I change to the way you just described?

Reply:
Ahh I didn't notice that, interesting. If you checkout the warning in [the XMLHttpRequest docs](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/onreadystatechange) though it says you shouldn't use these together so it would probably break on certain browsers. Also I know jQuery depricated synchronous AJAX requests because they're bad for user experience (they block the whole javascript thread) and you never know what unintentional bad behaviour it could have, so yeah, let's make it async and then just do my `else if` statement and I believe everything should be good